commcare_connect/program/helpers.py

```diff
@@ -0,0 +1,78 @@
+from django.db.models import (
+    Avg,
+    Case,
+    Count,
+    DurationField,
+    ExpressionWrapper,
+    F,
+    FloatField,
+    OuterRef,
+    Q,
+    Subquery,
+    Value,
+    When,
+)
+from django.db.models.functions import Cast, Round
+
+from commcare_connect.opportunity.models import UserVisit, VisitValidationStatus
+from commcare_connect.program.models import ManagedOpportunity, Program
+
+
+def calculate_safe_percentage(numerator, denominator):
```

Comment:
It's not worth for this to be a function, may be a private function inside the caller would be better